File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/domain/LastAddedRecipesUseCaseTest.java
package com.tarbadev.witchcraft.domain;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Java6Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(MockitoJUnitRunner.class)
public class LastAddedRecipesUseCaseTest {
  @Mock private RecipeRepository recipeRepository;

  private LastAddedRecipesUseCase subject;

  @Before
  public void setUp() {
    subject = new LastAddedRecipesUseCase(recipeRepository);
  }

  @Test
  public void execute() {
    List<Recipe> recipes = Arrays.asList(
        Recipe.builder().build(),
        Recipe.builder().build(),
        Recipe.builder().build(),
        Recipe.builder().build(),
        Recipe.builder().build()
    );

    given(recipeRepository.findLastAddedRecipes()).willReturn(recipes);

    assertThat(subject.execute()).isEqualTo(recipes);
  }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/recipes/domain/usecase/LastAddedRecipesUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import com.tarbadev.witchcraft.recipes.domain.entity.Recipe
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class LastAddedRecipesUseCaseTest {
    private val recipeRepository: RecipeRepository = mock()
    private lateinit var subject: LastAddedRecipesUseCase

    @BeforeEach
    fun setUp() {
        subject = LastAddedRecipesUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        val recipes = listOf(
            Recipe(),
            Recipe(),
            Recipe(),
            Recipe(),
            Recipe()
        )

        whenever(recipeRepository.findLastAddedRecipes()).thenReturn(recipes)

        assertEquals(recipes, subject.execute())
    }
}